system/jlib/jlog.hpp

```diff
@@ -1264,6 +1264,60 @@ interface jlib_decl IContextLogger : extends IInterface
     virtual const CRuntimeStatisticCollection & queryStats() const = 0;
 };
 
+template <class INTERFACE>
+class CDefaultContextLogger : public CInterfaceOf<INTERFACE>
+{
+public:
+    virtual void CTXLOGva(const LogMsgCategory & cat, const LogMsgJobInfo & job, LogMsgCode code, const char *format, va_list args) const __attribute__((format(printf,5,0))) = 0;
+    virtual void logOperatorExceptionVA(IException *E, const char *file, unsigned line, const char *format, va_list args) const __attribute__((format(printf,5,0))) = 0;
```

Comment:
No need to redeclare these pure virtuals, which the methods below call.
This template base class must be a IContextLogger derivative, and so the methods can use this-> (see below)